dev/com.ibm.ws.logging.osgi/src/com/ibm/ws/logging/internal/osgi/stackjoiner/ThrowableInfo.java

```diff
@@ -0,0 +1,93 @@
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.logging.internal.osgi.stackjoiner;
+
+import java.io.PrintStream;
+import java.lang.instrument.Instrumentation;
+import java.lang.reflect.Method;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+/**
+ * Retrieves the printStackTraceOverride method from BaseTraceService via reflection.
+ */
+public class ThrowableInfo {
+	
+	 private static final TraceComponent tc = Tr.register(ThrowableInfo.class);
+
+    final String BASE_TRACE_SERVICE_CLASS_NAME = "com.ibm.ws.logging.internal.impl.BaseTraceService";
+    final String BASE_TRACE_SERVICE_METHOD_NAME = "printStackTraceOverride";
+
+    private Method btsMethod;
+
+    /*
+    public ThrowableInfo(Instrumentation inst) {
+    	Class<?> btsClass = retrieveClass(inst, BASE_TRACE_SERVICE_CLASS_NAME);
+        if (btsClass != null) {
+			Method method = ReflectionHelper.getDeclaredMethod(btsClass, BASE_TRACE_SERVICE_METHOD_NAME, Throwable.class, PrintStream.class);
+			setBtsMethod(method);
+        }
+    }
+    
+    public boolean isInitialized() {
+    	if (getBtsMethod() == null) {
+    		if (tc.isDebugEnabled())
+    			Tr.debug(tc, "Stack joiner could not be initialized. Failed to reflect method " + BASE_TRACE_SERVICE_METHOD_NAME + " in " + BASE_TRACE_SERVICE_CLASS_NAME + ".");
+        	return false;
+        }
+    	return true;
+    }
+    */
+    
+    public ThrowableInfo(Instrumentation inst) {
+    	if (System.getenv("stackjoiner") != null && System.getenv("stackjoiner").equals("true")) {
+	    	Class<?> btsClass = retrieveClass(inst, BASE_TRACE_SERVICE_CLASS_NAME);
+	        if (btsClass != null) {
+				Method method = ReflectionHelper.getDeclaredMethod(btsClass, BASE_TRACE_SERVICE_METHOD_NAME, Throwable.class, PrintStream.class);
+				setBtsMethod(method);
+	        }
+    	}
+    }
+    public boolean isInitialized() {
+    	if (System.getenv("stackjoiner").equals("true")) {
+	    	if (getBtsMethod() == null) {
+	    		if (tc.isDebugEnabled())
+	    			Tr.debug(tc, "Stack joiner could not be initialized. Failed to reflect method " + BASE_TRACE_SERVICE_METHOD_NAME + " in " + BASE_TRACE_SERVICE_CLASS_NAME + ".");
+	        	return false;
+	        }
+	    	return true;
+    	}
+    	return false;
+    }
+    
+    @SuppressWarnings("rawtypes")
+	private Class<?> retrieveClass(Instrumentation inst, String classGroup) {
+        if (inst != null) {
+            Class[] loadedClasses = inst.getAllLoadedClasses();
```

Comment:
Class.forName in this instance will output a 'Class Not Found' error because the current class loader does not have the class that we need. 
By grabbing the classes from every class loader, we are then able to find the class that we need.